roles/monitoring_sap/tasks/main.yml

```diff
@@ -0,0 +1,37 @@
+---
+# This playbook checks, if SAP_HANA_CLIENT is insttalled  on the monitoring host 
+- name: Check existence of /usr/lib64/python3.6/site-packages/hdbcli/dbapi.py
+  stat:
+    path: /usr/lib64/python3.6/site-packages/hdbcli/dbapi.py
+  register: file_hana_driver
+
+- name: If SAP HANA client does not exist then install SAP HANA client
+  ansible.builtin.include_tasks: "{{ task }}.yml"
+  loop_control:
+    loop_var: task
+  loop:
+    - repos-activation        
+    - sap-hana-client-installation
+  when: not file_hana_driver.stat.exists and sap_monitoring_action == "add"
+    
+- name: Add one SAP System to the monitoring configuration
+  ansible.builtin.include_tasks: "{{ task }}.yml"
+  loop_control:
+    loop_var: task
+  loop:
+    - port-connectivity-check
+    - sap-host-exporter-configuration
+    - hanadb-exporter-configuration
+    - prometheus-agent-configuration
+  when: sap_monitoring_action == "add"
+
+##  Delete a SAP System from monitoring
+- name: Check existence of the variables defined in roles/monitoring_sap/vars/main.yml
```

Comment:
Move the tasks to a block

Reply:
block is a excellent idea,
the ansible docu jumps to Ansible 10 about block. Is it featured in Ansible 2.9?

Reply:
Yes supported in 2.9

Reply:
converted to 2 blocks, using the vars to control the action